src/pages/CustomerInvoicePage.js

```diff
@@ -75,6 +76,13 @@ export const CustomerInvoice = ({
     isFinalised,
   ]);
 
+  const runWithLoadingIndicator = useLoadingIndicator();
+
+  const onAddFromMasterLists = selected => {
+    onCloseModal();
+    runWithLoadingIndicator(() => onApplyMasterLists(selected, pageObject));
+  };
```

Comment:
pattern to be used in the rest of the PR: Close the modal before starting the spinner as that will more or less block the JS thread and it won't close the modal to show the spinner

Reply:
![thumbsup](https://user-images.githubusercontent.com/43223637/99922735-e7aab200-2d96-11eb-856e-9cbf62f36d79.gif)